pkg/fake/ssmapi.go

```diff
@@ -60,6 +67,86 @@ func (a SSMAPI) GetParameterWithContext(_ context.Context, input *ssm.GetParamet
 	}, nil
 }
 
+func (a SSMAPI) GetParametersByPathPagesWithContext(_ context.Context, input *ssm.GetParametersByPathInput, f func(*ssm.GetParametersByPathOutput, bool) bool, _ ...request.Option) error {
+	if a.WantErr != nil {
+		return a.WantErr
+	}
+	if a.GetParametersByPathOutput != nil {
+		f(a.GetParametersByPathOutput, true)
+		return nil
+	}
+	if len(a.Parameters) != 0 {
+		f(&ssm.GetParametersByPathOutput{
+			Parameters: lo.FilterMap(lo.Entries(a.Parameters), func(p lo.Entry[string, string], _ int) (*ssm.Parameter, bool) {
+				if !strings.HasPrefix(p.Key, lo.FromPtr(input.Path)) {
+					return nil, false
+				}
+				if strings.TrimPrefix(p.Key, lo.FromPtr(input.Path))[0] != '/' {
+					return nil, false
+				}
+				return &ssm.Parameter{
+					Name:  lo.ToPtr(p.Key),
+					Value: lo.ToPtr(p.Value),
+				}, true
+			}),
+		}, true)
+		return nil
+	}
+	if params := getDefaultParametersForPath(lo.FromPtr(input.Path)); params != nil {
+		f(&ssm.GetParametersByPathOutput{Parameters: params}, true)
+		return nil
+	}
+	return fmt.Errorf("path %q does not exist", lo.FromPtr(input.Path))
+}
+
+func getDefaultParametersForPath(path string) []*ssm.Parameter {
+	suffixes := map[string][]string{
+		`^\/aws\/service\/eks/optimized-ami\/.*\/amazon-linux-2$`: []string{"recommended/image_id"},
+		`^\/aws\/service\/eks/optimized-ami\/.*\/amazon-linux-2-arm64$`: []string{"recommended/image_id"},
+		`^\/aws\/service\/eks/optimized-ami\/.*\/amazon-linux-2-gpu$`: []string{"recommended/image_id"},
+		`^\/aws\/service\/eks/optimized-ami\/.*\/amazon-linux-2023$`: []string{
+			"x86_64/standard/recommended/image_id",
+			"arm64/standard/recommended/image_id",
+			"x86_64/nvidia/recommended/image_id",
+			"arm64/nvidia/recommended/image_id",
+			"x86_64/neuron/recommended/image_id",
+			"arm64/neuron/recommended/image_id",
+		},
+		`\/aws\/service\/bottlerocket\/aws-k8s-.*`: []string{
+			"x86_64/latest/image_id",
+			"arm64/latest/image_id",
+		},
+		`\/aws\/service\/ami-windows-latest`: lo.FlatMap(supportedK8sVersions(), func(version string, _ int) []string {
+			return []string{
+				fmt.Sprintf("Windows_Server-2019-English-Core-EKS_Optimized-%s/image_id", version),
+				fmt.Sprintf("Windows_Server-2022-English-Core-EKS_Optimized-%s/image_id", version),
+			}
+		}),
```

Comment:
There shouldn't be a cyclical dependency, but there also isn't a ton of value in defining these regexes / values in the AMI files. They're really only useful for this fake. 

The reason only Windows needs the k8s version injected is it's the only AMI which doesn't include the k8s version as part of the path hierarchy, but includes it directly in the image version. For all of the other AMIs we handle the k8s version is included in the path we send to `GetParametersByPath`, but not for Windows. This means for the other AMIs we can just append a k8s version independent set of suffixes to the path, but not for windows.

Reply:
Got it. I'm worried about us missing any newly added ssm paths in here if it's not some sort of shared const/var. What do you think we can do to prevent that from happening?

Reply:
Failing to add those should cause any tests relying on those paths to fail. I think that by itself solves the problem.